static/app/utils/profiling/flamegraphFrame.tsx

```diff
@@ -16,6 +18,14 @@ export interface FlamegraphFrame {
   threadId?: number;
 }
 
+export function getFlamegraphFrameDisplayName(frame: FlamegraphFrame) {
```

Comment:
Have we considered making just the name of the frame a "n collapsed frames" or even drop the n? I don't think many people care or if it's particularly useful to display how many frames were collapsed + they can always expand the tree to see which ones were collapsed.

Also something to note is that a frame may be collapsed for different reasons and controlling the name without adding more flags would just make it work downstream and we don't risk forgetting to call this fn everywhere.

Alternatively, we could make this a getter on the frame, although I dislike the idea of getters as we'll likely offload chart construction to workers at some point and having a serializable format would be nice

Reply:
getters is on the object is a bad idea w/o converting it to a class. I would avoid it

Reply:
```
changing the frame name; i've considered it, we'd have to create new instances of the call tree node to avoid mutating the original reference, but it would work
```
Oh right, bc the frame is just a reference right, we don't copy it. I think it's fine to clone it and override the name in that case - that also resembles the ds more closely as a collapsed frame is a new construct